application/src/org/commcare/entity/CommCareEntity.java

```diff
@@ -165,8 +186,42 @@ protected void loadEntity(TreeReference element) {
         loadTexts(ec);
     }
 
+    /**
+     * Given a detail, return an array of details that will contain either
+     * - all child details
+     * - a single-element array containing the given detail, if it has no children
+     * @param d The detail to flatten.
+     */
+    private Detail[] getFlattenedDetails(Detail d) {
```

Comment:
ha, I had the exact opposite thought, "This logic should live in J2ME because surely any other platform would support tabs." will update.